### PR TITLE
_WD_GetElementFromPoint()  - refactoring

### PR DESCRIPTION
**Pull request by mlipok**
#forceref $bFrame

### CODE REVIEW

**Comment by Danp2** [comment]
Can you explain why Au3check flags this variable? It's not making sense to me.

**Comment by Danp2** [comment]
I guess simply assigning a value to the variable isn't enough to pass the usage test. Originally, I had planned to include the following near the end of the function --
```
; Exit to top-most window
If $bFrame Then
	$sResponse = _WD_Window($sSession, "frame", '{"id":null}')
EndIf
```

That code is commented out because I believe the element ID being returned value outside of the current browsing context. I'm thinking it would be good to change last line to this so that the calling routine can check @extended to determine if the browsing context was changed --
```
	Return SetError($_WD_ERROR_Success, $bFrame, $sElement)
```

This would eliminate the need for the #forceref as well.

**Review by Danp2 — CHANGES_REQUESTED** [review]
Can you take a look at the changes that I just added to this PR and give your feedback?

**Comment by mlipok** [comment]
Yes in next few hours

**Review by Danp2 — APPROVED** [review]

**Comment by Danp2** [comment]
@mlipok Did you get a chance to review my changes?

**Comment by mlipok** [comment]
> @mlipok Did you get a chance to review my changes?

image

**Comment by Danp2** [comment]
@mlipok I think you need to finalize your review so that it is no longer `pending`

**Review by mlipok — COMMENTED** [review]
please check it

File: wd_helper.au3 [hunk 1]
```diff
@@ -416,27 +416,30 @@ EndFunc   ;==>_WD_GetMouseElement
...
 Func _WD_GetElementFromPoint($sSession, $iX, $iY)
 	Local $sResponse, $sElement, $oJSON, $sTagName, $sParams, $aCoords, $bFrame = False, $oERect
-	#forceref $bFrame
 	Local $sScript1 = "return document.elementFromPoint(arguments[0], arguments[1]);"
 	Local $sScript2 = "return new Array(window.pageXOffset, window.pageYOffset);"
+	Local $iErr = $_WD_ERROR_Success
 
 	While True
 		$sParams = $iX & ", " & $iY
 		$sResponse = _WD_ExecuteScript($sSession, $sScript1, $sParams)
```
[the remark this answers is not in the document]
Comment (Danp2) [reply wd_helper.au3:435]: Yes, I had considered adding additional error checking in relation to the calls to _WD_ExecuteScript. However, the idea was to also check for the JSON function returning an error as well if it doesn't find the designated node.

**Review by Danp2 — COMMENTED** [review]
Please take a look at the latest commit.

**Comment by mlipok** [comment]
> Please take a look at the latest commit.

Sure in next year... ;)

**Comment by mlipok** [comment]
Now it's your turn @Danp2

**Review by Danp2 — APPROVED** [review]

**Review by Danp2 — COMMENTED** [review]
RE: https://github.com/Danp2/WebDriver/pull/114/commits/36c8e6c94416fb1c021e631922d698802c8bbefa

AFAICS, this update doesn't fix or improve the functionality of the script. Seems like it's more of a coding style difference to me. 🙄

**Comment by mlipok** [comment]
> this update doesn't fix or improve the functionality of the script. Seems like it's more of a coding style difference to me.

almost yes, I also add comment to the code

**Comment by mlipok** [comment]
currently all this following snippets:

```
		If @error Then
			$iErr = $_WD_ERROR_RetValue
			ExitLoop
		EndIf

```
in this function, are placed at the same indentation.
It tidied up the loop processing a bit.

**Comment by mlipok** [comment]
Now it looks even better, so the code is easier to understand and maintain.
At least I hope so.

**Review by Danp2 — APPROVED** [review]

**Pull request merged**